### ISSUE TITLE
Hard to navigate docs front page, refguide link is missing

### ISSUE THREAD

**Issue by rgommers** [numpy/numpy#18419]
Looking at https://numpy.org/devdocs/, it took me quite a while to find the link to the single most important page: the API reference guide front page. There's a couple of issues:

1. Too many links:

<img width="401" alt="image" src="https://user-images.githubusercontent.com/98330/107986735-f9ce7200-6fcc-11eb-8f34-c8aa6ade3139.png">

2. Missing link to https://numpy.org/devdocs/reference/index.html (only can be reached from the header navbar, not obvious).

We're now using the same theme as Pandas, so why not borrow this beautiful and clear front page (https://pandas.pydata.org/docs/):

<img width="670" alt="image" src="https://user-images.githubusercontent.com/98330/107987006-86793000-6fcd-11eb-984e-f52f3a51d2ff.png">

**Comment by 8bitmp3** [comment]
Thanks @rgommers 👍 @melissawm @rossbar Should we start exploring this?

The Pandas docs layout, as @rgommers pointed out, looks like this:

<img width="1151" alt="image" src="https://user-images.githubusercontent.com/19637339/109550627-73835700-7ac7-11eb-9e7b-34e24a2f2537.png">

I wonder if the Sphinx theme supports a nav bar (similar to what we've got now) that you can include on the left-hand side:
<img width="358" alt="image" src="https://user-images.githubusercontent.com/19637339/109550730-91e95280-7ac7-11eb-8daa-e60bda2f2671.png">

If not, some of the links could go above or below the "large blocks": 

<img width="812" alt="image" src="https://user-images.githubusercontent.com/19637339/109550842-b6452f00-7ac7-11eb-9c49-55a54820fd19.png">

---

More inspiration where the layouts include nav bars + "large blocks": 

1) https://www.tensorflow.org/tutorials

<img width="1231" alt="image" src="https://user-images.githubusercontent.com/19637339/109550991-e55ba080-7ac7-11eb-8ad3-cb694a692ba6.png">

2) https://pytorch.org/tutorials/

<img width="1212" alt="image" src="https://user-images.githubusercontent.com/19637339/109550878-be9d6a00-7ac7-11eb-84d8-6616346ede82.png">

**Comment by rgommers** [comment]
> I wonder if the Sphinx theme supports a nav bar (similar to what we've got now) that you can include on the left-hand side:
image

We don't need those links, they're very messy and there's little point in preserving them.

> If not, some of the links could go above or below the "large blocks":

I'd suggest to start with the exact Pandas layout (including links like "mailing list" in the same format), and then see if we are missing anything. If each of the links currently on the front page has a natural place below one of the tiles, then we're all good. If not, we'll find a place for it. On first inspection, I think everything fits nicely.

**Comment by melissawm** [comment]
I love this idea. Maybe @marsbarlee can chip in, too, from the design point of view. But I agree with @rgommers , we can start with the Pandas layout and see if we miss anything.

**Comment by MarsBarLee** [comment]
Here's my mockup:
![panda page to Numpy without buttons cropped](https://user-images.githubusercontent.com/46167686/110983271-8d2e6500-8337-11eb-8329-8fb07b68b842.png)

Most of the text is taken straight from the pandas' page, just modified to NumPy. However, I've changed 'Developer guide' to 'Contributor's Guide' to emphasize the non-code contributions people can make.

The text above the blocks is from the [What is Numpy?](https://numpy.org/devdocs/user/whatisnumpy.html) page. 

---

This does make a nice 4 blocks, but when I referred to the pandas page, their blocks lead to more links. Here's my thoughts on how we can arrange ours:

**Getting Started**: On the pandas website, this leads to a [landing page with a few different guides](https://pandas.pydata.org/docs/getting_started/index.html#getting-started), such as different installations, introduction tutorials and for people familiar with R, SQL, etc.
We could make a similar landing page with NumPy [installation](https://numpy.org/install/), [absolute beginner's guide](https://numpy.org/devdocs/user/absolute_beginners.html), [F2PY user guide](https://numpy.org/devdocs/f2py/index.html), [NumPy for MATLAB users](https://numpy.org/devdocs/user/numpy-for-matlab-users.html) and [quickstart](https://numpy.org/devdocs/user/quickstart.html) ("Aimed at domain experts or people migrating to NumPy").

**User Guide**: Another landing page for NumPy [tutorials](https://numpy.org/devdocs/user/tutorials_index.html), [how-tos](https://numpy.org/devdocs/user/howtos_index.html) and [fundamentals](https://numpy.org/devdocs/user/basics.html).

**API Reference**: [NumPy Reference page](https://numpy.org/doc/stable/reference/).

**Contributor's Guide:** [Contributing to NumPy page](https://numpy.org/devdocs/dev/index.html).

I'm not sure where the other links, such as Building From Source would go.

**Comment by 8bitmp3** [comment]
@MarsBarLee 

<img width="200" alt="image" src="https://user-images.githubusercontent.com/19637339/110990818-c1237d80-836b-11eb-849a-6d57aa0c3533.png">

**Comment by 8bitmp3** [comment]
@MarsBarLee I'd personally suggest to minimize the number of clicks to 3 max before a user can find a relevant tutorial/guide. I read about this somewhere... maybe it was about AirBnb approaches its UX.

**Comment by rgommers** [comment]
> Most of the text is taken straight from the pandas' page, just modified to NumPy.

Makes sense - no need to rework things that are already well thought out:)

> However, I've changed 'Developer guide' to 'Contributor's Guide' to emphasize the non-code contributions people can make.

Good idea!

> User Guide: Another landing page for NumPy tutorials, how-tos and fundamentals.

I like all your four sections. The one thing I would add is: something like the "quickstart" can be both under "Getting Started" and under "User Guide". That makes navigation easier, there's no reason we have to choose where it goes cause it fits under both categories.

> I'd personally suggest to minimize the number of clicks to 3 max before a user can find a relevant tutorial/guide.

Agreed. That's how I read @MarsBarLee's plan already. E.g. User guide would have the direct links to tutorials, howto's and fundamentals. No need for another level of almost-empty pages like https://numpy.org/devdocs/user/howtos_index.html

**Comment by MarsBarLee** [comment]
Hi all, this is a mockup of the Getting Started page.
![Getting Started for NumPy-1](https://user-images.githubusercontent.com/46167686/114202724-7888c100-9925-11eb-9bd2-0ee6ddc7abd2.png)

Following pandas example, I've simplified the conda commands from the [installation page](https://numpy.org/install/) from
```
# Best practice, use an environment rather than install in the base env
conda create -n my-env
conda activate my-env
# If you want to install from conda-forge
conda config --env --add channels conda-forge
# The actual install command
conda install numpy
```
to
```
conda install numpy
```
I've also moved up the section on virtual environments to reduce repetition. The full installation commands can still be seen in the installation page.

The descriptions for the Starting Guides and 'Coming from...' are definitely still open for changes. I've pulled from their respective pages and summarize but if there's something wrong with the terminology, let me know!

Next up is making a mockup of the User Guide section. Eventually I'll put all the mockups on a clickable online prototype, so we can see how everything is linked together and keep to the '3 clicks max' rule.

**Comment by rgommers** [comment]
`matplotlib` may take offense at being mistaken for Matlab:) I suggest to just use regular text for `Matlab`, not a logo (that can have copyright issues). Other than that this looks quite good.

> Following pandas example, I've simplified the conda commands from the installation page from

I think our instructions are better - we added those on purpose. Users who need this kind of getting started info really need to be taught this.

**Comment by melissawm** [comment]
This looks great, @MarsBarLee !

A few points:
- there is an extra period at the end of `NumPy can be installed via pip:`
- For f2py, I'd suggest the last sentence to read "Learn about using f2py to wrap your code" or something similar. I don't think we should mention signature files here. 

Thanks!

**Comment by charris** [comment]
@melissawm Is there a PR in the works for this?

**Comment by melissawm** [comment]
Not yet - @MarsBarLee is working on a mockup first, and then she will tackle the actual implementation. But we're working on it!

**Comment by MarsBarLee** [comment]
Hi all, this is a link to the [interactive prototype](https://design.penpot.app/#/view/41890b30-934c-11eb-825c-a56ecc9e3fc1/e1ce3fc0-934c-11eb-bc52-e39ba1f3ae20?token=P8-a3msKJ73tEDHtfpC3Sg&index=0).
![insta](https://user-images.githubusercontent.com/46167686/118040543-a5bef980-b33f-11eb-9afb-1b19185fe4aa.gif)
This prototype exists as a low-cost, quick and iterative overview of how several pages could look and ‘feel’ together.

If you have a suggestion, if possible, put in your experience as context. For example- ‘As an experienced NumPy user, I would like to quickly reach the Reference. Could we put a link to the Reference in the left navigation bar?’
- In UI/UX terms, this is a ‘user profile’ or ‘user story’. It helps us understand and meet the needs of different users
- You can leave comments on Github instead of on Penpot (the prototyping platform)

Here's answers to some questions
- There is no direct ‘translation’ from prototype to code. Implementation will be through the current Sphinx theme
- This prototype is continually updated
- The advantage of prototyping is that instead of PRs that updates one page at a time, we can reach a consensus before putting a line of code, saving time down the road
- 'WYSIWYG'- no need for people download a branch and spin up a local instance. We can click this link and see
- Some pages are rougher/don't look at nice as others or missing links- that's also good feedback! I'm still updating some pages but as a whole I think this can be released for feedback

**Comment by isabela-pf** [comment]
Hey there @MarsBarLee! I know this is a delayed comment, but I've been thinking a little about this since you brought it up at the last NumPy docs meeting.

Most of this makes sense to me. My only note is that I find it confusing when the same image is used for different guides (even though they are on different pages). The only one I noticed was that "Getting Started" on the landing page and "Absolute Beginner's Guide" both use the running person.

**Comment by HaoZeke** [comment]
Minor nit about the Getting started design --> For F2PY though, the documentation doesn't really address (nor should it) moving from Fortran to NumPy.

I see the F2PY User and Development Guide to be at the same conceptual level as the User Guide or API Reference. Currently we have:
- User Guide
  + F2PY User Guide and Reference Guide

I'd suggest something along the lines of -->
- User
  + Blah blah
- F2PY
- API Docs
- Blah blah

**Comment by MarsBarLee** [comment]
Hi @isabela-pf and @HaoZeke, thank you for the feedback on re-organizing the content! I think for now, I'll complete the initial page layout that was laid out in the interactive prototype. The PR for that is #19756. Once that is done, I'll make a new PR for re-organizing the content.

I've also updated the prototype link- thanks Rohit for letting me know. A version update of the prototyping software changed the link.

**Comment by melissawm** [comment]
Hi folks, I think this is finalized for now, so I'm closing. Feel free to open new issues if you have further suggestions for improvements. Thanks!
